Cargo.toml

```diff
@@ -49,6 +49,9 @@ time = { version = "0.3", features = ["macros", "formatting", "parsing"] }
 ureq = { version = "2.5", default-features = false }
 rand = "0.8.5"
 rsa = { version = "0.8.1", features = ["sha2"] }
+urlencoding = "2.1.2"
+ring = "0.16.20"
```

Comment:
Why do we need to depend directly on `ring`? Can we use existing dependencies (by our best)?